### ISSUE TITLE
Revise and improve in-built converters

### ISSUE THREAD

**Issue by tombujok** [hazelcast/hazelcast#7591]
Currently the in-built converters are sub-optimal when it comes to performance. 
For example, if I want to convert a `Long` to using the `BigInteger` converter - it will first produce a string (litter) and then rely on the `BigInteger` parser to parse this String.
https://github.com/hazelcast/hazelcast/blob/eef987b9920f3f7b4c506e433b0d067cab9ba93c/hazelcast/src/main/java/com/hazelcast/query/impl/TypeConverters.java#L163

**Comment by tombujok** [comment]
Thanks @cpiotr, #8034 got merged!
Would you like to review other converters to see if there are any sub-optimal ones?

**Comment by cpiotr** [comment]
@tombujok: sure. Do you have any issues registered around that?

**Comment by tombujok** [comment]
No, just this github issue. I suspect there's similar sub-optimal stuff over there but never looked at all of them.

**Comment by tombujok** [comment]
Hi @cpiotr Did you have any spare time to hack on this ;) ?

**Comment by cpiotr** [comment]
Not yet. I'm currently on vacation. Will get down to it mid-June.

**Comment by cpiotr** [comment]
Hi @tombujok, I took a quick look at the rest of converters. There is a room for improvements in case of `java.sql.{Timestamp, Date}`. Converters for both classes rely on `DateHelper `which uses `SimpleDateFormat `to construct first `java.util.Date` and then instance of target type. 
In both cases, `valueOf `methods exist, which do the same job in shorter time:

Benchmark | Mode | Cnt | Score | Error | Units
--- | --- | --- | --- | --- | --- 
DateHelperTest.benchmarkDateHelper | avgt | 10 | 1.483 | 0.345 | us/op
DateHelperTest.benchmarkDateValueOf | avgt | 10 | 0.214 | 0.033 | us/op
DateHelperTest.benchmarkTimestampHelper | avgt | 10 | 2.047 | 0.517 | us/op
DateHelperTest.benchmarkTimestampValueOf | avgt | 10 | 0.454 | 0.093 | us/op

**Comment by mmedenjak** [comment]
Since we're heavily invested in the new query engine, I don't think it makes sense to put too much time into this.
